src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -27,223 +33,335 @@
-        this commonly refers to the month of the Current Situation period
-        if str, it should be in ISO 8601 format, e.g. '2021-10-01'
-    iso2 : str
-        iso2 code of the country of interest.
-        See https://fews.net/ for a list of countries that are
-        included by FewsNet
-    region_name : {'caribbean-central-america', 'central-asia',
-        'east-africa', 'southern-africa', 'west-africa'}
-        name of the region to which the `iso2` belongs.
-        Only regions for which FewsNet provides data can be given as input
-    region_code : {'LAC','EA','CA','SA','WA'}
-        code that refers to the `region_name`
-        These are defined by FewsNet
-    output_dir : Path or str
-        path to dir to which the data should be written
-    use_cache : bool, default True
-        if True, don't download if output_dir already exists
-
-    Examples
-    --------
-    >>> download_fewsnet(date_pub="2020-10-01",iso2="et",
-    ... region_code="east-africa", region_name="EA",
-    ... output_dir="tmp",use_cache=False)
+    country_config : CountryConfig
+        Country configuration
     """
-    if region_name not in _VALID_REGION_NAMES:
-        raise ValueError(f"Invalid region name {region_name}")
-    if region_code not in _VALID_REGION_CODES:
-        raise ValueError(f"Invalid region code {region_code}")
-
-    # convert to datetime if str
-    if not isinstance(date_pub, date):
-        date_pub = datetime.strptime(date_pub, _DATE_FORMAT)
-    # convert to path object if str
-    if isinstance(output_dir, str):
-        output_dir = Path(output_dir)
-
-    # we prefer the country data as this more nicely structured
-    # thus first check if that is available
-    country_path = _download_fewsnet_country(
-        date_pub=date_pub,
-        iso2=iso2,
-        output_dir=output_dir,
-        use_cache=use_cache,
-    )
-    if country_path is not None:
-        return country_path
-    else:
-        region_path = _download_fewsnet_region(
-            date_pub=date_pub,
-            region_name=region_name,
-            region_code=region_code,
-            output_dir=output_dir,
-            use_cache=use_cache,
+
+    def __init__(
+        self,
+        country_config,
+    ):
+        super().__init__(
+            country_config=country_config,
+            datasource_base_dir="fewsnet",
+            is_public=True,
+            # FN data can be regional, and thus we save the raw data
+            # in "glb" instead of in the iso3 folder, even though part of it is
+            # at country level
+            is_global_raw=True,
+            is_global_processed=False,
         )
-    if region_path is not None:
-        return region_path
-    else:
-        raise RuntimeError(f"No data found for {date_pub.strftime('%Y-%m')}")
 
+        self._iso2 = Country.get_iso2_from_iso3(self._country_config.iso3)
+        if self._iso2 is None:
+            raise KeyError(
```

Comment:
This file raises a lot of different errors for various situations, I would try to make sure those are covered by tests